addons/point_of_sale/static/src/app/store/models.js

```diff
@@ -1020,8 +1021,44 @@ export class Payment extends PosModel {
         return Boolean(this.get_payment_status());
     }
 
+    async showQR() {
+        const orm = this.env.services.orm
+        var qr
+        try {
+            try{
+                qr = await orm.call("pos.payment.method", "get_qr_code", [
+                    [this.payment_method.id],
+                    this.amount,
+                    this.order.name,
+                    this.order.name,
+                    this.pos.currency.id,
+                    this.order.partner?.id,
+                ]);
+            } catch {
+                qr = this.payment_method.default_qr
+            }
+            this.env.services.dialog.add(QRPopup, {
+                title: _t(this.name),
+                line: this,
+                order: this.order,
+                qrCode: qr,
+                close: () => {
+                    if (this.props.line.is_done()) {
+                        return true;
+                    } else { return false; }
+                }
+            })
+        } catch (error) {
+            console.log(error)
+            this.set_payment_status("pending");
+            return false
+        }
+    }
     async pay() {
         this.set_payment_status("waiting");
+        if (this.payment_method.use_qr) {
+            return await this.showQR()
```

Comment:
Right now the showQR will return undefined since there is no return statement. Do you need to make a promise to make it return the correct value from the qr code scanning?

Reply:
Yeah I think that would be better